app/assets/javascripts/terrier/tiny-modal.coffee

```diff
@@ -196,6 +202,10 @@ window.tinyModal.showDirect = (content, options={}) ->
 	column = $("<div class='modal-column'>#{fullContent}</div>").appendTo row
 	if options.columnClasses?.length
 		column.addClass tinyTemplate.parseClasses(options.columnClasses).join(' ')
+	if options.onShow?
+		callbackKey = Date().valueOf()
+		column.data('callback-id', callbackKey)
+		tinyModal.customCallbacks[callbackKey] = options.onShow
```

Comment:
(this comment is for line 216)

Your implementation is really a "onReshownAfterChildPopped". Besides that being a much worse name, I think it might be more intuitive to simply behave as an actual onShow - i.e. it gets called even the first time the modal is shown. Can you think of a reason not to do this? (having to modify your current usage in clypboard doesn't count)